File: clients/Cpp/model/Vec2Float.cpp
#include "Vec2Float.hpp"

Vec2Float::Vec2Float() { }
Vec2Float::Vec2Float(float x, float y) : x(x), y(y) { }
Vec2Float Vec2Float::readFrom(InputStream& stream) {
    Vec2Float result;
    result.x = stream.readFloat();
    result.y = stream.readFloat();
    return result;
}
void Vec2Float::writeTo(OutputStream& stream) const {
    stream.write(x);
    stream.write(y);
}

File: clients/TypeScript/src/model/Vec2Float.ts
import {Stream} from './Stream';

export class Vec2Float {
    constructor(
        private x: number,
        private y: number,
    ) {}

    static async readFrom(stream: Stream): Promise<Vec2Float> {
        const x = await stream.readFloat();
        const y = await stream.readFloat();
        return new Vec2Float(x, y);
    }

    async writeTo(stream: Stream): Promise<void> {
        await stream.writeFloat(this.x);
        await stream.writeFloat(this.y);
    }
}
